pomodoro-client.py

```diff
@@ -64,32 +65,51 @@ def format_pomodoro_data(pomodoro_data, icon_text, show_seconds):
 def format_output(pomodoro_data, always, icon_text, show_seconds):
 
     if pomodoro_data["state"] != "null":
-        return "{state} {remaining} {is_paused}".format(**format_pomodoro_data(
-            pomodoro_data, icon_text, show_seconds
-        ))
+        return "{state} {remaining} {is_paused}".format(
+            **format_pomodoro_data(pomodoro_data, icon_text, show_seconds)
+        )
     if always:
         return icon_text
     return ""
 
 
+def format_output_waybar(pomodoro_data, show_seconds):
+    import json
+
+    output = {"class": "stopped", "text": "", "tooltip": ""}
+    if pomodoro_data["state"] != "null":
+        data = format_pomodoro_data(pomodoro_data, show_seconds)
+        output["class"] = data["is_paused"].lower().strip()
+        output["text"] = data["remaining"]
+        output["tooltip"] = "Elapsed: {}\nRemaining: {}".format(
+            data["elapsed"], data["remaining"]
+        )
+    return json.dumps(output)
+
+
 @click.group()
 def main():
     pass
 
-@click.option('--always/--not-always', default=False,
-              help="""Show a constant icon.""")
-@click.option('--show-seconds/--no-seconds', default=True,
-              help="""Show seconds in timers""")
-@click.option('--icon-text', default="Pomodoro", help='What to show as icon.')
-@click.command(help=
-               """
+
+@click.option("--always/--not-always", default=False, help="""Show a constant icon.""")
+@click.option(
+    "--show-seconds/--no-seconds", default=True, help="""Show seconds in timers"""
+)
+@click.option("--waybar/--no-waybar", default=False, help="""Show waybar JSON output""")
+@click.option("--icon-text", default="Pomodoro", help="What to show as icon.")
+@click.command(
+    help="""
                Returns a string descriping the current pomodoro state.
-               """)
-def status(always, icon_text, show_seconds):
+               """
+)
+def status(always, icon_text, show_seconds, waybar):
     pomodoro = get_pomodoro_proxy()
     pomodoro_data = extract_pomodoro_data(pomodoro)
-    click.echo(format_output(pomodoro_data, always, icon_text, show_seconds))
-
+    if waybar:
```

Comment:
I am not familiar with waybar, so I cannot really comment on this, but maybe there's a better way of doing it than a flag?

It might be a somewhat larger re-factor, and I'm not sure what's the most efficient way of doing it without code duplication, but perhaps we could create 2 separate commands in this repo: `i3-gnome-pomodoro` and `waybar-gnome-pomodoro` :thinking:

It doesn't need to be part of this MR though

Reply:
Another idea is to create a `--format=` flag, which can take one of these values: `text` or `waybar`

Reply:
Can you please check if you think the same thing after my last git push force from the morning?
I would like to prevent duplicating and renaming the script this will be hard to maintain then.
I will try to transform "--waybar" in "--format",  better for later other format I think, good idea.
You can find some doc about the "custom" format and "return-type json" here https://github.com/Alexays/Waybar/wiki/Module:-Custom